packages/RadioGroup/src/index.tsx

```diff
@@ -36,31 +35,50 @@ export const RadioGroup = forwardRef<'fieldset', RadioGroupProps>(
       options = [],
       renderOption: Component = Radio,
       required,
+      dataTestId,
+      onChange,
       value,
       ...rest
     },
     ref
   ) => {
-    const radio = useRadioState({ currentId: value })
-    const withHint = options.findIndex(obj => Object.keys(obj).includes('hint')) !== -1
+    const store = Ariakit.useRadioStore({ defaultValue: value })
+    const activeValue = store.useState('value')
+
+    const handleChange = (valueSelected: Option['value']) => {
+      if (rest.disabled) return
+
+      store.setValue(valueSelected)
+      onChange?.(valueSelected)
+    }
+
+    const withHint = options.findIndex((obj: Option) => Object.keys(obj).includes('hint')) !== -1
```

Comment:
J'ai l'impression qu'on veut retourner `true` dès qu'on trouve l'attribut `hint`, on pourrait utiliser `some` qui nous permettrait de pas forcément à avoir à itérer sur toutes les options : 

```ts
const withHint = options.some((obj: Option) => "hint" in obj)
```

Reply:
Ah mais en fait je viens de voir que cette ligne a pas été rajoutée mais juste déplacée donc pas forcément à appliquer ici :)